src/collectors/definitions.cpp

```diff
@@ -26,7 +26,11 @@ class DefinitionsDataAppender : public MatchFinder::MatchCallback {
     clang::SourceManager *sm = r.SourceManager;
     const clang::FileID fid = sm->getFileID(e->getLocation());
     const clang::FileEntry *entry = sm->getFileEntryForID(fid);
+#if LLVM_VERSION_MAJOR >= 15
+    if (!entry) {
+#else
     if (!(entry && entry->isValid())) {
+#endif
```

Comment:
Seems that `FileEntry *` are being moved away from towards `OptionalFileEntryRef`

Can also change line 28 to: Use `clang::SourceManager::getFileEntryRefForID` https://clang.llvm.org/doxygen/classclang_1_1SourceManager.html#a474c5b69c25c5664f8624b965424f04d